File: csc13001-plant-pos.Core/Models/OrderDetail.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace csc13001_plant_pos.Core.Models;

[Table("order_details")]
public class OrderDetail
{
    [Key]
    [Column("order_detail_id")]
    public int OrderDetailId
    {
        get; set;
    }

    [Column("order_id")]
    [ForeignKey("Order")]
    public int OrderId
    {
        get; set;
    }
    public Order Order
    {
        get; set;
    }


    [Column("product_id")]
    [ForeignKey("Product")]
    public int ProductId
    {
        get; set;
    }
    public Product Product
    {
        get; set;
    }

    [Column("quantity")]
    public int Quantity
    {
        get; set;
    }

    [Column("unit_price", TypeName = "decimal(10,2)")]
    public decimal UnitPrice
    {
        get; set;
    }

    [Column("discount_id")]
    [ForeignKey("Discount")]
    public int? DiscountId
    {
        get; set;
    }
    public DiscountProgram? Discount
    {
        get; set;
    }
}

File: backend/src/main/java/csc13001/plantpos/domain/models/OrderDetail.java
package csc13001.plantpos.domain.models;

import jakarta.persistence.*;
import jakarta.validation.constraints.Min;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "order_details")
public class OrderDetail {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "order_detail_id")
    private Long orderDetailId;

    @Column(name = "order_id")
    private Long orderId;

    @Column(name = "product_id")
    private Long productId;

    @Min(value = 1, message = "Quantity must be at least 1")
    @Column(name = "quantity")
    private int quantity;

    @Column(name = "unit_price", columnDefinition = "decimal(18,2)")
    private BigDecimal unitPrice;

    @Column(name = "discount_id")
    private Long discountId;
}
